### PR TITLE
[BACKLOG-36722] Fixed keyboard navigation for CustomListBox

### PR DESCRIPTION
**Pull request by AAravind27**
@pentaho/wcag , Please Review

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-901/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/listbox/CustomListBox.java [hunk 1]
```diff
@@ -1140,6 +1124,11 @@ public void itemSelected( ListItem listItem, Event event ) {
     } else {
       setSelectedItem( listItem );
     }
+
+    if ( visible == 1 && popupShowing ) { // Drop-down mode
+      togglePopup();
+    }
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/listbox/CustomListBox.java:1130]: Please, do not add comments at the end of the line. Above or inside the `if` would be fine.

File: widgets/src/test/java/org/pentaho/gwt/widgets/client/listbox/CustomListBoxTest.java [hunk 2]
```diff
@@ -234,7 +234,6 @@ public void testSetSelectedItem() throws Exception {
     when( customListBox.items.indexOf( item ) ).thenReturn( selectionIndex );
     customListBox.setSelectedItem( item );
     verify( selectedItem ).onDeselect();
-    verify( customListBox ).togglePopup();
```
Comment (dcleao) [inline widgets/src/test/java/org/pentaho/gwt/widgets/client/listbox/CustomListBoxTest.java:237]: A test already exists for the `itemSelected( . )` method. Maybe you could add this assertion there, or create a new test of not appropriate.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 42s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-901/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by hv-vamsi — APPROVED** [review]

**Pull request merged**